statistics.cpp

```diff
@@ -10,14 +11,15 @@
 // This function solves the ODE based on odeModel, odeParameters, and solverParameters
 // Then the ODE solution and data are converted to histograms based on histogramParameters.
 // Finally the log likelihood is computed using the multinomial distribution and the two histograms.
-double Statistics::logLikelihood(const std::valarray<double>& particleSizeData,
-                                 const Models::ModelsBase& odeModel,
-                                 const Models::ParametersBase& odeParameters,
-                                 const Models::explEulerParameters& solverParameters,
-                                 const Histograms::Parameters& histogramParameters)
+std::tuple<double, std::vector<std::valarray<double>>, std::vector<std::valarray<double>>> Statistics::logLikelihoodInternal(const std::vector<std::valarray<double>>& particleSizeData,
```

Comment:
you might want to document what this awkward return object actually is

Reply:
This change to the loglikelihood function is also conceptually separate from the remaining changes. You could put this into a separate PR as well and see what the tests have to say about it.